specification/search/Azure.Search/models-index.tsp

```diff
@@ -0,0 +1,1077 @@
+   * the field names of each fielded search expression take precedence over any
+   * field names listed in this parameter.
+   */
+  searchFields?: string;
+
+  /**
+   * A value that specifies whether any or all of the search terms must be matched
+   * in order to count the document as a match.
+   */
+  searchMode?: SearchMode;
+
+  /**
+   * The comma-separated list of fields to retrieve. If unspecified, all fields
+   * marked as retrievable in the schema are included.
+   */
+  select?: string;
+
+  /**
+   * The number of search results to skip. This value cannot be greater than
+   * 100,000. If you need to scan documents in sequence, but cannot use skip due to
+   * this limitation, consider using orderby on a totally-ordered key and filter
+   * with a range query instead.
+   */
+  skip?: int32;
+
+  /**
+   * The number of search results to retrieve. This can be used in conjunction with
+   * $skip to implement client-side paging of search results. If results are
+   * truncated due to server-side paging, the response will include a continuation
+   * token that can be used to issue another Search request for the next page of
+   * results.
+   */
+  top?: int32;
+
+  /**
+   * The name of a semantic configuration that will be used when processing
+   * documents for queries of type semantic.
+   */
+  semanticConfiguration?: string;
+
+  /**
+   * Allows the user to choose whether a semantic call should fail completely
+   * (default / current behavior), or to return partial results.
+   */
+  semanticErrorHandling?: SemanticErrorMode;
+
+  /**
+   * Allows the user to set an upper bound on the amount of time it takes for
+   * semantic enrichment to finish processing before the request fails.
+   */
+  @minValue(700)
+  semanticMaxWaitInMilliseconds?: int32;
+
+  /**
+   * Allows setting a separate search query that will be solely used for semantic
+   * reranking, semantic captions and semantic answers. Is useful for scenarios
+   * where there is a need to use different queries between the base retrieval and
+   * ranking phase, and the L2 semantic phase.
+   */
+  semanticQuery?: string;
+
+  /**
+   * A value that specifies whether answers should be returned as part of the search
+   * response.
+   */
+  answers?: QueryAnswerType;
+
+  /**
+   * A value that specifies whether captions should be returned as part of the
+   * search response.
+   */
+  captions?: QueryCaptionType;
+
+  /**
+   * The query parameters for vector and hybrid search queries.
+   */
+  vectorQueries?: VectorQuery[];
+
+  /**
+   * Determines whether or not filters are applied before or after the vector search
+   * is performed. Default is 'preFilter' for new indexes.
+   */
+  vectorFilterMode?: VectorFilterMode;
+}
+
+/**
+ * The query parameters for vector and hybrid search queries.
+ */
+#suppress "@azure-tools/typespec-azure-core/no-string-discriminator" "Existing"
+@discriminator("kind")
+model VectorQuery {
```

Comment:
I'm not seeing the "kind" property on this model which is also leading to that warning you had to suppress.